### PR TITLE
Code cleanup/optimization, variables renaming for clarity

### PR DESCRIPTION
**Pull request by ttngu207**

### CODE REVIEW

**Review by shenshan — APPROVED** [review]

File: element_array_ephys/__init__.py [hunk 1]
```diff
@@ -1,3 +1,33 @@
...
+    # search routine
+    for root_dir in potential_root_directories:
+        root_dir = pathlib.Path(root_dir)
+        if path.exists():
+            if root_dir in list(path.parents):
+                return path, root_dir
+        else:
+            if (root_dir / path).exists():
+                return root_dir / path, root_dir
+
```
Comment (dimitri-yatsenko) [inline element_array_ephys/__init__.py:31]:
```suggestion
        if path.exists() or (root_dir / path).exists():
            return path if root_dir in set(path.parents) else root_dir / path, root_dir
```

File: element_array_ephys/readers/kilosort.py [hunk 2]
```diff
@@ -64,52 +64,64 @@ def info(self):
...
-            self._data['cluster_ids'] = np.array(df['cluster_id'].values)
+        for cluster_pattern, cluster_col_name in zip(['cluster_groups.*', 'cluster_KSLabel.*'],
+                                                     ['group', 'KSLabel']):
+            try:
+                cluster_file = next(self._ks_dir.glob(cluster_pattern))
+                cluster_file_suffix = cluster_file.suffix
+                assert cluster_file_suffix in ('.csv', '.tsv', '.xlsx')
+                break
+            except StopIteration:
+                pass
```
Comment (dimitri-yatsenko) [inline element_array_ephys/readers/kilosort.py:102]:
```suggestion
            try:
                cluster_file = next(self._ks_dir.glob(cluster_pattern))
            except StopIteration:
                pass
            cluster_file_suffix = cluster_file.suffix
            assert cluster_file_suffix in ('.csv', '.tsv', '.xlsx')
            break

```

File: element_array_ephys/readers/kilosort.py [hunk 3]
```diff
@@ -64,52 +64,64 @@ def info(self):
...
+        self._data['cluster_ids'] = np.array(df['cluster_id'].values)
 
     def get_best_channel(self, unit):
         template_idx = self.data['spike_templates'][
             np.where(self.data['spike_clusters'] == unit)[0][0]]
-        chn_templates = self.data['templates'][template_idx, :, :]
-        max_chn_idx = np.abs(np.abs(chn_templates).max(axis=0)).argmax()
-        max_chn = self.data['channel_map'][max_chn_idx]
+        channel_templates = self.data['templates'][template_idx, :, :]
+        max_channel_idx = np.abs(np.abs(channel_templates).max(axis=0)).argmax()
```
Comment (dimitri-yatsenko) [inline element_array_ephys/readers/kilosort.py:121]:
```suggestion
        max_channel_idx = np.abs(channel_templates).max(axis=0).argmax()
```

**Review by dimitri-yatsenko — CHANGES_REQUESTED** [review]

File: element_array_ephys/readers/kilosort.py [hunk 4]
```diff
@@ -64,52 +64,66 @@ def info(self):
...
-            self._data['cluster_ids'] = np.array(df['cluster_id'].values)
+        for cluster_pattern, cluster_col_name in zip(['cluster_groups.*', 'cluster_KSLabel.*'],
+                                                     ['group', 'KSLabel']):
+            try:
+                cluster_file = next(self._kilosort_dir.glob(cluster_pattern))
+            except StopIteration:
+                pass
+            else:
+                cluster_file_suffix = cluster_file.suffix
+                assert cluster_file_suffix in ('.csv', '.tsv', '.xlsx')
```
Comment (dimitri-yatsenko) [inline element_array_ephys/readers/kilosort.py:103]: Add message to assert. Or raise a specific error

**Review by dimitri-yatsenko — APPROVED** [review]

**Pull request merged**